### PR TITLE
Folders: Smart component integration with RPC

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

This PR:

* sets the 'new' and 'ignored' metas on folders coming in from the favoriteList RPC, so that our dumb component (which already knew what to do with those) badges them appropriately

* makes the "Ignore folder" button perform a favoriteIgnore RPC on that folder

* after a favoriteIgnore RPC, go back to the folders list and refresh the favorites list

No visdiff changes expected, since it's all data integration on the smart component.

### CODE REVIEW

File: shared/constants/favorite.js [hunk 1]
```diff
@@ -6,4 +6,7 @@ import type {Props as FolderProps} from '../folders/render'
 export const favoriteList = 'favorite:favoriteList'
 export type FavoriteList = TypedAction<'favorite:favoriteList', {folders: FolderProps}, void>
 
-export type FavoriteAction = FavoriteList
+export const favoriteIgnore = 'favorite:favoriteIgnore'
+export type FavoriteIgnore = TypedAction<'favorite:favoriteIgnore', {error: ?string}, void>
```
Comment (MarcoPolo) [inline shared/constants/favorite.js]: Why not use `error: true` ?

https://github.com/keybase/client/blob/master/shared/constants/types/flux.js#L3

File: shared/actions/favorite.js [hunk 2]
```diff
@@ -72,8 +97,9 @@ const folderToProps = (folders: Array<Folder>, username: string = ''): FolderPro
     return a.sortName.localeCompare(b.sortName)
   })
 
-  const [priv, pub] = _.partition(converted, {isPublic: false})
-
+  const [priFolders, pubFolders] = _.partition(converted, {isPublic: false})
+  let [privIgnored, priv] = _.partition(priFolders, {ignored: true})
+  let [pubIgnored, pub] = _.partition(pubFolders, {ignored: true})
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: why not `const`

**Comment by cjb** [comment]
@MarcoPolo Thanks! Fixed both, PTAL.

**Comment by chrisnojima** [comment]
- [ ] Clicking on the 'Open' button on hover should open the finder
- [ ] Clicking on ignored doesn't open the files view so i can't unignore

File: shared/actions/favorite.js [hunk 3]
```diff
@@ -1,57 +1,82 @@
...
 
 export function pathFromFolder ({isPublic, users}: {isPublic: boolean, users: UserList}) {
   const sortName = users.map(u => u.username).join(',')
-  const path = `/keybase/${isPublic ? 'private' : 'public'}/${sortName}`
+  const path = `/keybase/${isPublic ? 'public' : 'private'}/${sortName}`
   return {sortName, path}
 }
 
-const folderToProps = (folders: Array<Folder>, username: string = ''): FolderProps => { // eslint-disable-line space-infix-ops
+export function folderFromPath (path: string): ?Folder {
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: Why are we exporting this?

Comment (MarcoPolo) [inline shared/actions/favorite.js]: Also seems a bit weird, but if it's local to this file I guess it's okay

File: shared/actions/favorite.js [hunk 4]
```diff
@@ -161,6 +194,30 @@ export function favoriteList (): (dispatch: Dispatch) => void {
   }
 }
 
-export function ignoreFolder (path: string) {
-  return () => console.log('TODO: implement ignore folder action')
+export function ignoreFolder (path: string): (dispatch: Dispatch) => void {
+  return (dispatch, getState) => {
+    const folder = folderFromPath(path)
+    if (!folder) {
+      const action: FavoriteIgnore = {type: Constants.favoriteIgnore, payload: {error: true}}
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: Sorry I wasn't clear, I meant adding the error flag up one level, like this: https://github.com/keybase/client/blob/master/shared/actions/signup/index.js#L47

and then for the non error case:

https://github.com/keybase/client/blob/master/shared/actions/signup/index.js#L50

File: shared/constants/favorite.js [hunk 5]
```diff
@@ -6,4 +6,7 @@ import type {Props as FolderProps} from '../folders/render'
 export const favoriteList = 'favorite:favoriteList'
 export type FavoriteList = TypedAction<'favorite:favoriteList', {folders: FolderProps}, void>
 
-export type FavoriteAction = FavoriteList
+export const favoriteIgnore = 'favorite:favoriteIgnore'
+export type FavoriteIgnore = TypedAction<'favorite:favoriteIgnore', {error?: true}, void>
```
Comment (MarcoPolo) [inline shared/constants/favorite.js]: This second type argument is the non error payload, the third is the error payload. I think these would both be void.

File: shared/actions/favorite.js [hunk 6]
```diff
@@ -161,6 +194,30 @@ export function favoriteList (): (dispatch: Dispatch) => void {
...
+    }
+
+    const params : favoriteFavoriteIgnoreRpc = {
+      param: {folder},
+      incomingCallMap: {},
+      method: 'favorite.favoriteIgnore',
+      callback: error => {
+        if (error) {
+          console.warn('Err in favorite.favoriteIgnore', error)
+          return
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: If there is an error, are you stuck somewhere? since you aren't navigating back?

**Comment by cjb** [comment]
> Clicking on ignored doesn't open the files view so i can't unignore

Looks like this was intentional -- we don't have a folder object for ignoredFiles so the view can't render them.  (It also wouldn't be able to "open folder"; the folder doesn't exist until you've unignored it.)

Do we have any design for unignoring?  Should it be done from a hover on the ignored folders list, or from the files view?

**Comment by chrisnojima** [comment]
We do have the folder objects from the rpc layer already, we're likely just not mapping them. All folders exist, if you go there it'll exist. ignoring just means removing it from favorites which removes it from 'ls /keybase/private|public' but it still exists if you cd /keybase/private/chrisnojima,blah

File: shared/actions/favorite.js (hunk 3 above)
Comment (cjb) [inline shared/actions/favorite.js]: thanks, unexported

File: shared/actions/favorite.js (hunk 4 above)
Comment (cjb) [inline shared/actions/favorite.js]: thanks for the help, done

**Comment by cjb** [comment]
@chrisnojima @MarcoPolo made all of your changes now, PTAL!  These both work now:

* Clicking on the 'Open' button on hover should open the finder
* Clicking on ignored doesn't open the files view so i can't unignore

**Comment by chrisnojima** [comment]
can we put that padding on top: https://zpl.io/Z1008aU
feels weird with the tabs that high up

**Comment by chrisnojima** [comment]
Is there no plain ignore button?

**Comment by chrisnojima** [comment]
lets make the ignore page like the normal page, so 2 buttons 'unignore' and 'open'.
let's also hide the menu on files page based on the feature flag

then 👍

**Comment by MarcoPolo** [comment]
👍

**Comment by cjb** [comment]
@chrisnojima Made your two changes (explicit unignore button, hide menubar on feature flag), thanks.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Files-Recent_Files_Disabled_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466102550135-4a27df08a3e7-60cf39a2edfa/Files-Recent_Files_Disabled_Public-4a27df08a3e7.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466102550135-4a27df08a3e7-60cf39a2edfa/Files-Recent_Files_Disabled_Public-60cf39a2edfa.png)  
   Files-Recent_Files_Disabled_Public comparison
 * Changed: **Files-Recent_Files_Disabled_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466102550135-4a27df08a3e7-60cf39a2edfa/Files-Recent_Files_Disabled_Private-4a27df08a3e7.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466102550135-4a27df08a3e7-60cf39a2edfa/Files-Recent_Files_Disabled_Private-60cf39a2edfa.png)  
   Files-Recent_Files_Disabled_Private comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Menubar-Public_Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Public_Truncated-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Public_Truncated-ea685b6384a1.png)  
   Menubar-Public_Truncated comparison
 * Changed: **Menubar-Public_Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Public_Normal-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Public_Normal-ea685b6384a1.png)  
   Menubar-Public_Normal comparison
 * Changed: **Menubar-Public_Menu_Showing** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Public_Menu_Showing-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Public_Menu_Showing-ea685b6384a1.png)  
   Menubar-Public_Menu_Showing comparison
 * Changed: **Menubar-Private_Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Truncated-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Truncated-ea685b6384a1.png)  
   Menubar-Private_Truncated comparison
 * Changed: **Menubar-Private_Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Normal-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Normal-ea685b6384a1.png)  
   Menubar-Private_Normal comparison
 * Changed: **Menubar-Private_Menu_Showing** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Menu_Showing-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Menu_Showing-ea685b6384a1.png)  
   Menubar-Private_Menu_Showing comparison
 * Changed: **Folders_TLF-Normal_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Folders_TLF-Normal_Public-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Folders_TLF-Normal_Public-ea685b6384a1.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **Folders_TLF-Normal_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Folders_TLF-Normal_Private-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Folders_TLF-Normal_Private-ea685b6384a1.png) [(diff)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Folders_TLF-Normal_Private.png)
 * Changed: **Files-Recent_Files_Disabled_Public** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Files-Recent_Files_Disabled_Public-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Files-Recent_Files_Disabled_Public-ea685b6384a1.png) [(diff)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Files-Recent_Files_Disabled_Public.png)
 * Changed: **Files-Recent_Files_Disabled_Private** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Files-Recent_Files_Disabled_Private-60cf39a2edfa.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Files-Recent_Files_Disabled_Private-ea685b6384a1.png) [(diff)](http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Files-Recent_Files_Disabled_Private.png)

**Comment by cjb** [comment]
(This visdiff is expected, added paddingTop to match Zeplin.)

**Comment by chrisnojima** [comment]
This is actually incorrect. the padding shouldn't be that large in the menu widget. you can see here http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Menu_Showing-60cf39a2edfa.png vs http://keybase-app-visdiff.s3.amazonaws.com/1466103700449-60cf39a2edfa-ea685b6384a1/Menubar-Private_Menu_Showing-ea685b6384a1.png

**Comment by cjb** [comment]
Doh, thanks.  Opened #3121 with a fix.

**Pull request merged**